alchemy/src/planetscale/role.ts

```diff
@@ -0,0 +1,240 @@
+   */
+  id: string;
+
+  /**
+   * The name of the role
+   */
+  name: string;
+
+  /**
+   * The timestamp when the role expires (ISO 8601 format)
+   */
+  expiresAt: string;
+
+  /**
+   * The host URL for database connection
+   */
+  host: string;
+
+  /**
+   * The username for database authentication
+   */
+  username: string;
+
+  /**
+   * The encrypted password for database authentication
+   */
+  password: Secret<string>;
+
+  /**
+   * The database name
+   */
+  databaseName: string;
+
+  /**
+   * The direct connection URL for the database.
+   */
+  connectionUrl: Secret<string>;
+
+  /**
+   * The pooled connection URL for the database.
+   * Uses PSBouncer on port 6432. Recommended for production.
+   * @see https://planetscale.com/docs/postgres/connecting/psbouncer
+   */
+  connectionUrlPooled: Secret<string>;
+}
+
+/**
+ * Create and manage database roles for PlanetScale PostgreSQL branches. Database roles provide secure access to your database with specific roles and permissions.
+ *
+ * For MySQL, use [Passwords](./password.ts) instead.
+ *
+ * @example
+ * ## Basic Role
+ *
+ * Create a default role with all permissions:
+ *
+ * ```ts
+ * const role = await Role("my-role", {
+ *   database: "my-database",
+ *   inheritedRoles: ["postgres"],
+ * });
+ * ```
+ *
+ * ## Role with TTL
+ *
+ * Create a role with a TTL of 1 hour:
+ *
+ * ```ts
+ * const role = await Role("my-role", {
+ *   database: "my-database",
+ *   ttl: 3600,
+ * });
+ * ```
+ *
+ * ## Role with Inherited Permissions
+ *
+ * Create a role with read-only access to all data and settings:
+ *
+ * ```ts
+ * const role = await Role("my-role", {
+ *   database: "my-database",
+ *   inheritedRoles: ["pg_read_all_data", "pg_read_all_settings"],
+ * });
+ * ```
+ */
+export const Role = Resource(
+  "planetscale::Role",
+  async function (
+    this: Context<Role, RoleProps>,
+    _id: string,
+    props: RoleProps,
+  ): Promise<Role> {
+    const api = new PlanetScaleClient(props);
+    const organization =
+      typeof props.branch === "object"
+        ? props.branch.organizationId
+        : typeof props.database === "object"
+          ? props.database.organizationId
+          : props.organizationId;
+    const database =
+      typeof props.database === "string" ? props.database : props.database.name;
+    const branch =
+      typeof props.branch === "string"
+        ? props.branch
+        : (props.branch?.name ?? "main");
+    if (!organization) {
+      throw new Error("Organization ID is required");
+    }
+
+    switch (this.phase) {
+      case "delete": {
+        if (this.output?.id) {
+          const res = await api.organizations.databases.branches.roles.delete({
+            path: {
+              organization,
+              database,
+              branch,
+              id: this.output.id,
+            },
+            result: "full",
+          });
+          // TODO: (422 unprocessable): Role is still referenced and cannot be dropped
```

Comment:
Any idea what to do about this error?

If you delete the database or branch then the role is deleted automatically so this shouldn't technically be blocking.

Reply:
That looks fatal. Or is it something we need to wait for?

For now let's just crash and let users report issues.